react/src/components/ImageList.tsx

```diff
@@ -0,0 +1,345 @@
+      key: 'constraint',
+      dataIndex: 'constraint',
+      sorter: (a, b) => {
+        const [, , ...requirementsA] = a?.tag?.split('-') || ['', '', ''];
+        const [, , ...requirementsB] = b?.tag?.split('-') || ['', '', ''];
+        const requirementA = requirementsA.join('');
+        const requirementB = requirementsB.join('');
+        if (requirementA === '' && requirementB === '') return 0;
+        if (requirementA === '') return -1;
+        if (requirementB === '') return 1;
+        return requirementA.localeCompare(requirementB);
+      },
+      render: (text, row) => (
+        <ConstraintTags
+          image={getImageFullName(row) || ''}
+          labels={row?.labels as { key: string; value: string }[]}
+        />
+      ),
+    },
+    {
+      title: t('environment.Digest'),
+      dataIndex: 'digest',
+      key: 'digest',
+      sorter: (a, b) =>
+        a?.digest && b?.digest ? a.digest.localeCompare(b.digest) : 0,
+    },
+    {
+      title: t('environment.ResourceLimit'),
+      dataIndex: 'resource_limits',
+      key: 'resource_limits',
+      render: (text, row) =>
+        row?.resource_limits?.map((resource_limit) => (
+          <ResourceNumber
+            // @ts-ignore
+            type={resource_limit.key}
+            value={resource_limit?.min || '0'}
+            max={resource_limit?.max || ''}
+          />
+        )),
+    },
+    {
+      title: t('general.Control'),
+      key: 'control',
+      dataIndex: 'control',
+      fixed: 'right',
+      render: (text, row) => {
+        return (
+          <>
+            <Flex direction="row" align="stretch" justify="center" gap="xxs">
+              <Button
+                type="text"
+                icon={
+                  <SettingOutlined
+                    style={{
+                      color: token.colorInfo,
+                    }}
+                  />
+                }
+                onClick={() => setManagingResourceLimit(row)}
+              />
+              <Button
+                type="text"
+                icon={
+                  <AppstoreOutlined
+                    style={{
+                      color: token.colorInfo,
+                    }}
+                  />
+                }
+                onClick={() => {
+                  setManagingApp(row);
+                }}
+              />
+            </Flex>
+          </>
+        );
+      },
+    },
+  ];
+
+  return (
+    <>
+      <Flex justify="end" style={{ height: '3rem', padding: '.5rem 1rem' }}>
+        <Button
+          size="middle"
+          icon={<VerticalAlignBottomOutlined />}
+          style={{ backgroundColor: token.colorPrimary, color: 'white' }}
+          onClick={() => {
+            setIsOpenInstallModal(true);
+          }}
+        >
+          {t('environment.Install')}
+        </Button>
+      </Flex>
+      <Table
+        rowKey="id"
+        scroll={{ x: '100vw', y: 'calc(100vh - (9.0625rem + 3rem + 4.5rem))' }}
```

Comment:
```suggestion
scroll={{ x: 'max-content' }}
```

It seems to be affecting the styling of the table columns. The following modifications should fix the issue.